File: javascript/problem097.js
var result = 28433;

for(i=1; i<=7830457; i++){
  result = result * 2;
  var str = result.toString();
  if(str.length > 10){
    str = str.substring(1,11);
    result = parseInt(str);
  }
}

result = result + 1;

console.log(result);

File: java/src/Problem097.java
public class Problem097 {

    long result() {
        long value = 28433;
        for (int i = 1; i <= 7830457; i++) {
            value = value * 2;
            String valueStr = Long.toString(value);
            if (valueStr.length() > 10) {
                String sub = valueStr.substring(1, 11);
                value = Long.parseLong(sub);
            }
        }
        value = value + 1;
        return value;
    }

    public static void main(String args[]) {
        Problem097 problem = new Problem097();
        System.out.println(problem.result());
    }
}